### PR TITLE
sort

### PR DESCRIPTION
**Pull request by jtlap**

### CODE REVIEW

File: include/eve/module/core/regular/sort.hpp [hunk 1]
```diff
@@ -7,10 +7,26 @@
...
+#include <eve/traits/overload.hpp>
+#include <eve/module/core/decorator/core.hpp>
 
 namespace eve
 {
+  template<typename Options>
+  struct sort_t : callable<sort_t, Options>
+  {
+    template<eve::value T, typename Less>
+    constexpr EVE_FORCEINLINE T operator()(T v, Less l) const noexcept
```
Comment (DenisYaroshevskiy) [inline include/eve/module/core/regular/sort.hpp:20]: There should be a concept for less. Like maybe `eve::strict_weak_ordering<T>`
I'm not sure where that one should be declared.

Also probably not value but simd_value, I don't think scalars work

Comment (jtlap) [reply include/eve/module/core/regular/sort.hpp:20]: ok

Comment (DenisYaroshevskiy) [reply include/eve/module/core/regular/sort.hpp:20]: I guess? Also should be the same as std for scalar probably?

Comment (jfalcou) [reply include/eve/module/core/regular/sort.hpp:20]: yean it's a bit complicated. What fi we fix that in another PR ?

**Review by DenisYaroshevskiy — APPROVED** [review]

**Pull request merged**